sfm_pc/base_views.py

```diff
@@ -113,17 +113,8 @@ class BaseDeleteView(DeleteView):
     """
     def get_context_data(self, *args, **kwargs):
         context = super().get_context_data(*args, **kwargs)
-        context['cancel_link'] = self.get_cancel_link()
         return context
 
-    def get_cancel_link(self):
-        """
-        Return the route that the Cancel button should link to in the Delete template.
-        For example, if the user is editing an Organization Composition, this method
-        should return the route for Edit page for the Composition in question.
-        """
-        raise NotImplementedError('This method must be implemented on children.')
```

Comment:
In `edit.html` I am using conditional logic to check for the existence of the `get_cancel_url` function. It seemed to be the most reliable way to ensure it was there, so removing these abstract classes was necessary. Is there a better convention to follow here?

Reply:
I was thinking of the conditional template logic as a temporary measure to preserve backwards-compatibility while we migrate the rest of the views to use `get_cancel_url()`. If that's the case, then having a required abstract interface for `get_cancel_url()` seems useful to me, in that it will complain loudly if a developer ever tries to create a new view and forgets to implement a cancel URL (in contrast to the conditional check, which will fail silently). The only situation where it would seem to me to be appropriate to not require `get_cancel_url()` would be if we will wind up wanting Create/Edit/Delete views that don't have a cancel button, which doesn't seem to be true at this point.

From a more nuts-and-bolts perspective, I'm also surprised that child classes were throwing `NotImplementedError`. `BaseDeleteView` is a fairly new class so I thought that all children implemented the cancel link. Is it possible the error was getting raised because we changed the method name to `get_cancel_url` on the base class without changing the children?